### PR TITLE
patch request completed

### PR DESCRIPTION
**Pull request by Hartooro**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -188,4 +188,58 @@ describe('4 get/api/articles/:article_id', () => {
...
+        return request(app).patch("/api/articles/1").send(newVote).expect(200).then((res)=>{
+            expect(res.body.article).toEqual(expect.objectContaining(
+                {
+                    article_id :1,
+                    title : "Living in the shadow of a great man",
+                    topic: "mitch",
+                    author: "butter_bridge",
+                    body : "I find this existence challenging",
+                    created_at : "2020-07-09T20:11:00.000Z",
+                    votes : 50
```
Comment (xViolaine) [inline __tests__/app.test.js:221]: good job also checking if the decrease works

File: __tests__/app.test.js [hunk 2]
```diff
@@ -188,4 +188,58 @@ describe('4 get/api/articles/:article_id', () => {
...
+            title : "Coding bootcamp for dummies" }
+        return request(app).patch("/api/articles/1").send(newVote).expect(400).then((res)=>{
+            expect(res.body.msg).toEqual("you can only edit the votes. Nothing else!")
+        })
+    })
+    test('Very similar to previous test, but only attempts to change one property, not votes ', () => {
+        const newVote = {
+            title : "Coding bootcamp for dummies" }
+        return request(app).patch("/api/articles/1").send(newVote).expect(400).then((res)=>{
+            expect(res.body.msg).toEqual("you can only edit the votes. Nothing else!")
```
Comment (xViolaine) [inline __tests__/app.test.js]: Some very creative tests!
lets think of some more: 
What happens if they send an empty object? Or if inc_votes is not a number?
what if the id isn't existent, OR invalid?

**Review by xViolaine — CHANGES_REQUESTED** [review]
tl;dr: more tests!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -232,14 +232,40 @@ describe('4 get/api/articles/:article_id', () => {
...
+            expect(res.body.msg).toEqual("Bad, bad Request")
+        })
+    })
+    test('rejects 400 if a user tries to patch with an empty object', () => {
+        const newVote = { }
+         return request(app).patch("/api/articles/1").send(newVote).expect(400).then((res)=>{
+             expect(res.body.msg).toEqual("You can only edit the votes.")
+         })
+    })
+    test('rejects if user inserts invalid article id or non numerical', () => {
```
Comment (MKD01) [inline __tests__/app.test.js]: This test description is not accurate. This test is if the article id is valid but does not exist.

**Review by MKD01 — APPROVED** [review]
Happy for this to be merged after those changes have been made.
